var T=(G,D,u)=>new Promise((w,A)=>{var B=f=>{try{N(u.next(f))}catch(M){A(M)}},P=f=>{try{N(u.throw(f))}catch(M){A(M)}},N=f=>f.done?w(f.value):Promise.resolve(f.value).then(B,P);N((u=u.apply(G,D)).next())});import"./index.65159add.js";import{_ as L}from"./index.5a6dfef3.js";import{_ as X,u as q}from"./table.b14e82ad.js";import"./el-input.172ae120.js";/* empty css                  */import{u as Z}from"./saveScrollPosition.7fb4c1aa.js";import{g as J,u as Y,a as tt}from"./loading.150065e6.js";import{d as K,a as y,r as C,e as et,f as at,o as p,B as ot,w as a,k as n,i as o,t as j,h as e,g as m,q as $,F as z,s as nt,j as x,a8 as lt,a9 as st,E as it,aa as rt}from"./index.27a20cdf.js";import{a as H,d as E}from"./index.eddab6a4.js";import{a as dt,g as ct}from"./index.cb30be27.js";import{_ as pt}from"./plugin-vue_export-helper.db4a78ed.js";var ut="/hgmt/static/help.c76bf006.jpg",ft="/hgmt/static/HelpFigure3.e4d5e2f9.png",mt="/hgmt/static/HelpFigure1.9336eb87.png",gt="/hgmt/static/HelpFigure2.2fa9a690.png";const bt={style:{"list-style-type":"disc","margin-left":"3px","padding-left":"5px"}},ht={style:{display:"-webkit-box","text-overflow":"ellipsis",overflow:"hidden","-webkit-line-clamp":"3","-webkit-box-orient":"vertical"}},xt={key:0,style:{"list-style-type":"disc","margin-left":"3px","padding-left":"5px"}},yt={key:0,style:{"list-style-type":"disc","margin-left":"3px","padding-left":"5px"}},wt={key:1},vt=K({__name:"dataTable",setup(G){const D=y(0);Z(D);const u=C({data:[],total:0,page:1,limit:10,keywords:"",except:!1,totalData:[]}),w=[{label:"Project ID",slotName:"pid",prop:"project_id",width:"200",sortable:"custom",sortOrders:["descending","ascending"]},{label:"Assay Type",slotName:"assayType",prop:"assay_type",width:"120",sortable:!1},{label:"Nr. Processed Runs (Total Runs)",slotName:"runs",prop:"processed_runs",width:"160",sortable:!1},{label:"Related Phenotype(s)",slotName:"phenotypes",prop:"related_phenotypes",width:"360"},{label:"Publication",slotName:"publication",prop:"public_accession",width:"160"}],A=C({}),B=y([]),P=y([]),N=C({}),{handleSortChange:f,changePage:M,getTableData:V,searchTable:O,refreshSearch:W}=q(B,u);J(B,P,A,N,O);const v=g=>{window.open(g)},Q=g=>{g.indexOf("/")!==-1?v(`https://doi.org/${g}`):v(`https://www.ncbi.nlm.nih.gov/pubmed/?term=${g}`)};return(g,_)=>{const I=L,S=et,R=at,F=X;return p(),ot(F,{onPaginationChange:x(M),"table-data":x(u).data,"table-head":w,total:x(u).total,"default-sort":{prop:"valid_runs",order:"descending"},onSortChange:x(f)},{pid:a(({row:l})=>[n(I,{to:`/data/${l.project_id}`},{default:a(()=>[o(j(l.project_id),1)]),_:2},1032,["to"]),n(R,{content:"click to view it in NCBI BioProject",effect:"dark",placement:"top"},{default:a(()=>[n(S,{type:"primary",size:"small",style:{float:"right"},onClick:d=>v("https://www.ncbi.nlm.nih.gov/bioproject/"+l.project_id)},{default:a(()=>_[0]||(_[0]=[o(" NCBI\xA0 "),e("i",{class:"iconfont icon-share white"},null,-1)])),_:2},1032,["onClick"])]),_:2},1024)]),assayType:a(({row:l})=>{var d;return[e("ul",bt,[(p(!0),m(z,null,$((d=l.assay_type)==null?void 0:d.split(";"),k=>(p(),m("li",{key:k,style:{display:"list-item"}},j(k),1))),128))])]}),title:a(({row:l})=>[n(R,{content:l.project_title,placement:"top"},{default:a(()=>[e("p",ht,j(l.project_title),1)]),_:2},1032,["content"])]),runs:a(({row:l})=>[o(j(l.processed_runs)+" ("+j(l.total_runs)+") ",1)]),phenotypes:a(({row:l})=>[l.related_phenotypes!=="NA"?(p(),m("ul",xt,[(p(!0),m(z,null,$(l.related_phenotype_ids.toString().split(";"),(d,k)=>(p(),m("li",{style:{display:"flex","justify-content":"space-between","align-items":"center",padding:"3px"},key:d},[n(I,{to:`/explore/${d}`},{default:a(()=>[o(j(l.related_phenotypes.toString().split(";")[k]),1)]),_:2},1032,["to"]),n(R,{content:"click to view it in Mesh",effect:"dark",placement:"top"},{default:a(()=>[n(S,{type:"primary",size:"small",onClick:c=>v("https://meshb.nlm.nih.gov/record/ui?ui="+d)},{default:a(()=>_[1]||(_[1]=[o(" MeSH\xA0 "),e("i",{class:"iconfont icon-share white"},null,-1)])),_:2},1032,["onClick"])]),_:2},1024)]))),128))])):nt("",!0)]),publication:a(({row:l})=>[l.public_accession!=="NA"?(p(),m("ul",yt,[(p(!0),m(z,null,$(l.public_accession.toString().split(","),d=>(p(),m("li",{style:{display:"list-item"},key:d},[n(I,{to:"_blank","is-external":"","show-icon":!1,onClick:k=>Q(d)},{default:a(()=>[o(j(d)+" ",1),_[2]||(_[2]=e("i",{class:"iconfont icon-share"},null,-1))]),_:2},1032,["onClick"])]))),128))])):(p(),m("span",wt,"NA"))]),_:1},8,["onPaginationChange","table-data","total","onSortChange"])}}});const _t={class:"px-10 mx-auto md:w-screen-md lg:w-screen-lg xl:w-screen-xl help-page"},kt={class:"demo-collapse"},Tt=K({__name:"index",setup(G){const D=y("1"),u=C({}),w=y([]),A=y([]),B=C({}),P=C({data:[],total:0,page:1,limit:10,keywords:"",except:!1,totalData:[]}),{handleSortChange:N,changePage:f,getTableData:M,searchTable:V,refreshSearch:O}=q(w,P);J(w,A,u,B,V);const W=()=>T(this,null,function*(){const c=w.value;let t=`${["ID","Project ID","Assay Type","Project Title","Project Description","Public Accession","Journal"].join("	")}
`;c.forEach(s=>{const i=[];for(const h in s)i.push(s[h]);t+=i.join("	")+`
`});const b=new Blob([t],{type:"text/plain;charset=utf-8"}),r="all_projects_"+Date.now().toString()+".txt";H(b,r)}),v=C({data:[],total:1600,page:1,limit:10}),{loading:Q,toggle:g}=Y();(()=>{g(),tt({limit:1e3,skip:0}).then(c=>{v.data=c,g()})})();const I=()=>T(this,null,function*(){const c=v.data;let t=`${["ID","Project ID","Run ID","Sample Name","Assay Type","Sequencing Method","Phenotype Name","Sample Description","Phenotype ID","Country","Geographic Location","Longitude","Lattitude","Age","Sex","BMI","antibiotic_use","antibiotic_name","bodysite","batch","batch2","QC_Bacteria","QC_Bacteria_gg","QC_Fungi","QC_state"].join("	")}
`;c.forEach(s=>{const i=[];for(const h in s)i.push(s[h]);t+=i.join("	")+`
`});const b=new Blob([t],{type:"text/plain;charset=utf-8"}),r="all_runs_"+Date.now().toString()+".txt";H(b,r)}),S=y([]);function R(){return T(this,null,function*(){S.value=yield dt()})}const F=()=>T(this,null,function*(){yield R();const c=S.value;let t=`${["id","da_id","batch","taxa","kingdom","taxon_level","w_statistic","p_value","lfc","assay_type","case_name","control_name","num_case","num_control","taxa_short_name","display "," nrproj "," conflict"].join("	")}
`;c.forEach(s=>{const i=[];for(const h in s)i.push(s[h]);t+=i.join("	")+`
`});const b=new Blob([t],{type:"text/plain;charset=utf-8"}),r="DA_results_"+Date.now().toString()+".txt";H(b,r)}),l=y([]);function d(){return T(this,null,function*(){l.value=yield ct()})}const k=()=>T(this,null,function*(){yield d();const c=l.value;let t=`${["id","hgmt_micro_id","level","taxa","short_name","taxid","scientific_name"].join("	")}
`;c.forEach(s=>{const i=[];for(const h in s)i.push(s[h]);t+=i.join("	")+`
`});const b=new Blob([t],{type:"text/plain;charset=utf-8"}),r="taxa2ncbi_"+Date.now().toString()+".txt";H(b,r)});return(c,t)=>{const b=lt,r=st,s=it,i=L,h=rt;return p(),m("div",_t,[n(b,{class:"box-card text-2xl font-semibold",shadow:"hover"},{default:a(()=>t[1]||(t[1]=[e("span",{class:"text-pink-100"}," Help ",-1)])),_:1}),e("div",kt,[n(h,{modelValue:D.value,"onUpdate:modelValue":t[0]||(t[0]=U=>D.value=U),accordion:""},{default:a(()=>[n(r,{name:"1"},{title:a(()=>t[2]||(t[2]=[e("div",{class:"text-lg font-bold"}," 1. What is HGMT? ",-1)])),default:a(()=>[t[3]||(t[3]=e("div",null,[o(" HGMT is a comprehensive web server of gut microbiota from pan-tumor patients, which contains taxonomic and functional profiles annotated with state-of-the-art tools. Our efforts have led to the standardized processing of "),e("span",{class:"text-red-800 font-bold italic"},"18,630"),o(" treatment-naive human fecal samples, including "),e("span",{class:"text-red-800 font-bold italic"},"45"),o(" WGS, "),e("span",{class:"text-red-800 font-bold italic"},"74"),o(" 16S, and "),e("span",{class:"text-red-800 font-bold italic"},"3"),o(" ITS from "),e("span",{class:"text-red-800 font-bold italic"},"111"),o(" projects of "),e("span",{class:"text-red-800 font-bold italic"},"37"),o(" tumor types. Particularly, HGMT defines scores to evaluate the ability of gut microbiota for tumor diagnosis (GMTD) and tumor prognosis (GMTP). ")],-1)),t[4]||(t[4]=e("br",null,null,-1))]),_:1}),n(r,{name:"2"},{title:a(()=>t[5]||(t[5]=[e("div",{class:"text-lg font-bold"}," 2. What can HGMT provide? ",-1)])),default:a(()=>[t[6]||(t[6]=e("div",null,[e("dl",null,[e("dd",null," manual curation of metadata on tumor-related metagenomes from recently published studies; "),e("dd",null," calculation of multi-kingdom microbial abundance using state-of-the-art tools, i.e., Bacteria-kingdom and Fungi-kingdom; "),e("dd",null," assignment of the assembly-based functional profile, i.e., KO abundance; "),e("dd",null," evaluation of gut microbiota applied in tumor diagnosis (GMTD) and tumor prognosis (GMTP). ")])],-1)),t[7]||(t[7]=e("div",{class:"single-img"},[e("img",{src:ut,alt:"figure1"}),e("span",{class:"font-bold"},"Overview of the HGMT workflow")],-1))]),_:1}),n(r,{name:"3"},{title:a(()=>t[8]||(t[8]=[e("div",{class:"text-lg font-bold"}," 3. Which datasets are included in this database? ",-1)])),default:a(()=>[n(vt)]),_:1}),n(r,{name:"4"},{title:a(()=>t[9]||(t[9]=[e("div",{class:"text-lg font-bold"}," 4. What are the functions of the 'Explore' page ",-1)])),default:a(()=>[t[10]||(t[10]=e("div",null,[o(" The 'Explore' page provides interactive functions for "),e("dl",null,[e("dd",null," evaluation of gut microbiota in tumor diagnosis (GMTD) and tumor prognosis (GMTP), e.g., CRC diagnosis; and "),e("dd",null," a phenotype table summarizing differential abundance taxa of each tumor.")]),e("br"),e("div",{class:"imgbox single-img"},[e("img",{src:ft,alt:"figure4"})]),e("br"),e("br"),o(" \xA0\xA0For each phenotype in the table, HGMT provides two hyperlinks: "),e("i",{class:"iconfont icon-share"}),o(" external hyperlink (link to NCBI MeSH Browsers) and "),e("i",{class:"iconfont icon-link"}),o(" internal hyperlink (link to the 'marker taxa' page of a certain tumor within the 'Explore' module for cross-dataset disease marker comparison). For instance, users can click on the 'View details "),e("i",{class:"iconfont icon-link"}),o("' button of CRC to overview all differential abundance marker taxa across CRC-related projects (1). "),e("br"),e("br"),o(" \xA0\xA0For the 'marker taxa' page of a certain tumor type (e.g., CRC) within the 'Explore' module, users can overview disease marker taxa of all CRC-related projects or processed batches (2). Meanwhile, the tile view for each tumor phenotype within the 'Explore' module provides the interactive visualization for cross-dataset disease marker taxa comparison under different parameter settings, such as Assay_type, Kingdom, taxonomic level, and 'log2(FoldChange)' cutoff (3). To quickly attain consistent marker taxa considering occurrences and enrichment directions across projects, HGMT also provides two additional parameters: 'Nr.projects' cutoff and 'Exclude conflicting markers' (3). The tile view only exhibits marker taxa meeting chosen screening criteria, while the table view will save all differential abundance test results and allow users to download them (4). "),e("br"),e("br"),e("div",{class:"imgbox single-img"},[e("img",{src:mt,alt:"figure4"})])],-1))]),_:1}),n(r,{name:"5"},{title:a(()=>t[11]||(t[11]=[e("div",{class:"text-lg font-bold"}," 5. What are the functions of the 'Data' page? ",-1)])),default:a(()=>[t[12]||(t[12]=e("div",null,[o(" The 'Data' page provides the summary information of all projects and runs. For each project, HGMT provides two hyperlinks: "),e("i",{class:"iconfont icon-share"}),o(" external hyperlink (link to NCBI BioProject Browser) and "),e("i",{class:"iconfont icon-link"}),o(" internal hyperlink (link to the 'project' page of each project within 'Data' module, where users can download taxonomic and functional profiles of gut microbiota). Additional links to PubMed/bioRxiv (related publications), and NCBI SRA (Run ID) Browsers have also been contained. For example, users can click on 'PRJEB6070' "),e("i",{class:"iconfont icon-link"}),o(" internal hyperlink to the 'project' page (1). "),e("br"),e("br"),o(" \xA0\xA0For the 'project' page of one certain project (e.g., PRJEB6070) within the 'Data' module, users can overview the summary information of PRJEB6070 (2). Meanwhile, HGMT provides the meta and feature tables of associated runs based on different processed batches and kingdoms (3-4). The 'marker taxa' module allows users to link to cross-dataset disease maker taxa comparison for each tumor-health pair (5) and explore all marker taxa for each DA batch visualized by the barplot (6). ")],-1)),t[13]||(t[13]=e("div",{class:"imgbox single-img"},[e("img",{src:gt,alt:"figure10"})],-1))]),_:1}),n(r,{name:"6"},{title:a(()=>t[14]||(t[14]=[e("div",{class:"text-lg font-bold"}," 6. How do I download the data? ",-1)])),default:a(()=>[e("div",null,[t[26]||(t[26]=o(" Users can download the information of all projects and runs including metadata and feature tables in the 'Data' module. The 'Explore' module allows to download all differential abundance taxa for each tumor. Meanwhile, users can selectively search for interested projects or diseases by inputting keywords on the top navigation bar, which will link to the corresponding detailed page to download specific results. ")),t[27]||(t[27]=e("br",null,null,-1)),t[28]||(t[28]=e("dt",null,"Here, we offer a faster way as follows:",-1)),e("dl",null,[e("dd",null,[t[16]||(t[16]=o(" download the summary data of all projects:\xA0\xA0")),n(i,{"is-external":"","show-icon":!1,onClick:W},{default:a(()=>[t[15]||(t[15]=o("download")),n(s,null,{default:a(()=>[n(x(E))]),_:1})]),_:1})]),e("dd",null,[t[18]||(t[18]=o(" download the metadata of all runs;\xA0\xA0")),n(i,{"is-external":"","show-icon":!1,onClick:I},{default:a(()=>[t[17]||(t[17]=o("download")),n(s,null,{default:a(()=>[n(x(E))]),_:1})]),_:1})]),e("dd",null,[t[20]||(t[20]=o(" download the feature table of all processed runs:\xA0\xA0 ")),n(i,{"is-external":"","show-icon":!1,href:"/hgmt/src/assets/data/featureTable.zip"},{default:a(()=>[t[19]||(t[19]=o(" download")),n(s,null,{default:a(()=>[n(x(E))]),_:1})]),_:1}),t[21]||(t[21]=e("dl",null,[e("dd",null,"Bacterial profiles annotated by MetaPhlAn3 or QIIME2;"),e("dd",null,"Fungal profiles annotated by Kraken2+Bracken or QIIME2;"),e("dd",null,"KO profiles annotated by eggNOG-mapper;")],-1))]),e("dd",null,[t[23]||(t[23]=o("download the list of all disease marker taxa;\xA0\xA0")),n(i,{"is-external":"","show-icon":!1,onClick:F},{default:a(()=>[t[22]||(t[22]=o("download")),n(s,null,{default:a(()=>[n(x(E))]),_:1})]),_:1})]),e("dd",null,[t[25]||(t[25]=o("download the reformatted NCBI taxonomy table, including genera and species in HGMT.\xA0\xA0")),n(i,{"is-external":"","show-icon":!1,onClick:k},{default:a(()=>[t[24]||(t[24]=o("download")),n(s,null,{default:a(()=>[n(x(E))]),_:1})]),_:1})])])])]),_:1}),n(r,{name:"7"},{title:a(()=>t[29]||(t[29]=[e("div",{class:"text-lg font-bold"}," 7. How to cite HGMT? ",-1)])),default:a(()=>[t[30]||(t[30]=e("div",null,null,-1))]),_:1}),n(r,{name:"8"},{title:a(()=>t[31]||(t[31]=[e("div",{class:"text-lg font-bold"}," 8. How to contact us? ",-1)])),default:a(()=>[t[32]||(t[32]=e("div",null," For any questions or specific demands, please contact us. (Prof. Xing-Ming Zhao: xmzhao@fudan.edu.cn). ",-1))]),_:1})]),_:1},8,["modelValue"])])])}}});var Ht=pt(Tt,[["__scopeId","data-v-30ca9fcf"]]);export{Ht as default};
