/* empty css             */import"./plugin-vue_export-helper.21578988.js";import{_ as m}from"./plugin-vue_export-helper.db4a78ed.js";import{d as p,r as _,E as f,a3 as h,a4 as g,o as x,e as v,f as e,h as o,w as a,g as n,u as w,F as b,p as y,k}from"./index.cc3e9507.js";import{i as I}from"./info-filled.b02f316d.js";const t=l=>(y("data-v-01f0f79c"),l=l(),k(),l),E=t(()=>e("h2",{class:"mx-auto md:w-screen-md lg:w-screen-lg xl:w-screen-xl",style:{"text-align":"center","font-size":"32px",width:"100px",color:"#fff",background:"#fd6802","border-radius":"8px","padding-top":"15px"}},[e("span",null,"Help")],-1)),H={class:"demo-collapse"},T={class:"text-lg"},M=t(()=>e("div",{class:"info"}," HGMT is a comprehensive database of the multi-kingdom gut microbiota from pan-tumor patients. We uniformly processed a total of 16,195 treatment-naive human stool samples, including 35 WGS, 67 16S and 3 ITS from 99 projects of 32 tumor types. ",-1)),C=t(()=>e("div",{class:"text-lg"}," 2. What can HGMT provide? ",-1)),S=t(()=>e("div",{class:"info"},[e("dd",null,[e("dl",null,"(1) manual curation of metadata on tumor-related metagenomes from recently published studies; "),e("dl",null," (2) calculation of multi-kingdom microbial abundance using state-of-the-art tools; "),e("dl",null," (3) generation of tumor-related microbial landscape. ")])],-1)),W=t(()=>e("div",{class:"text-lg"}," 3. Which datasets are included in this database? ",-1)),V=t(()=>e("div",{class:"text-lg"}," 4. What are the functions of the 'Explore' page ",-1)),j=t(()=>e("div",{class:"info"},[n(" The 'Explore' page provides search functions for "),e("dl",null,"(1)Statistics; "),e("dl",null,"(2)Cross-dataset biomarker taxa comparison in a certain tumor type...")],-1)),B=t(()=>e("div",{class:"text-lg"}," 5. What are the functions of the 'Data' page? ",-1)),F=t(()=>e("div",{class:"info"}," The 'Data' page mainly provides Projects/processed runs summary information, Each project has an introduction page, where can download gut microbiota abundance tables and link to cross-dataset biomarker taxa comparison on the 'Explore' page. ",-1)),G=t(()=>e("div",{class:"text-lg"}," 6. How do I download the data? ",-1)),N=t(()=>e("div",{class:"info"},[n(" Users can download all projects and runs information including metadata and feature tables on the 'Data' page; The 'Explore' page allows to download cross-dataset marker taxa by differential abundance test. Here, we offer a faster way as follows:"),e("br"),e("dl",null,"(1) dowload all projects summary data:"),e("dl",null,"(2) dowload metadata of all associated runs;"),e("dl",null,"(3) dowload featdata of all associated runs: Bacteria profilles annotated by MetaPhlAn3/QIIME2; Fungi profilles annotated by Kraken2+Bracken/QIIME2;"),e("dl",null,"(4) dowload all biomarker taxa list.")],-1)),D=t(()=>e("div",{class:"text-lg"}," 7. How to cite HGMT? ",-1)),P=t(()=>e("div",{class:"info"},null,-1)),z=t(()=>e("div",{class:"text-lg"}," 8. How to contact us? ",-1)),Q=t(()=>e("div",{class:"info"}," For any question or specific demand, please contact us. (Prof. Xing-Ming Zhao: xmzhao@fudan.edu.cn). ",-1)),U=p({__name:"index",setup(l){const d=_("1");return(q,i)=>{const c=f,s=h,r=g;return x(),v(b,null,[E,e("div",H,[o(r,{modelValue:d.value,"onUpdate:modelValue":i[0]||(i[0]=u=>d.value=u),accordion:""},{default:a(()=>[o(s,{name:"1"},{title:a(()=>[e("div",T,[n(" 1. What is HGMT? "),o(c,{class:"header-icon"},{default:a(()=>[o(w(I))]),_:1})])]),default:a(()=>[M]),_:1}),o(s,{name:"2"},{title:a(()=>[C]),default:a(()=>[S]),_:1}),o(s,{name:"3"},{title:a(()=>[W]),_:1}),o(s,{name:"4"},{title:a(()=>[V]),default:a(()=>[j]),_:1}),o(s,{name:"5"},{title:a(()=>[B]),default:a(()=>[F]),_:1}),o(s,{name:"6"},{title:a(()=>[G]),default:a(()=>[N]),_:1}),o(s,{name:"7"},{title:a(()=>[D]),default:a(()=>[P]),_:1}),o(s,{name:"8"},{title:a(()=>[z]),default:a(()=>[Q]),_:1})]),_:1},8,["modelValue"])])],64)}}});var L=m(U,[["__scopeId","data-v-01f0f79c"]]);export{L as default};
